### PR TITLE
Feature/sckan 323

### PR DESCRIPTION
**Pull request by afonsobspinto**
Closes https://metacell.atlassian.net/browse/SCKAN-323

https://github.com/user-attachments/assets/f56e0026-9e68-4ab0-b2dc-d3761f6f7e31

As a draft because I need to test some edge cases (updates on many to many, like vias and destinations)

### CODE REVIEW

File: backend/composer/api/permissions.py [hunk 1]
```diff
@@ -1,9 +1,108 @@
...
+        # Allow system user to bypass all checks
+        if request.user.username == 'system' and request.user.is_staff:
+            return True
+
+        # Allow read-only access (GET, HEAD, OPTIONS)
+        if request.method in permissions.SAFE_METHODS:
+            return True
+
+        # For POST (create), PUT, PATCH (update), or DELETE, check ownership
+        return check_related_entity_ownership(request)
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply backend/composer/api/permissions.py:79]: This is used on the creation of ConnectivityStatements. When creating a connectivity statement via the API with a POST we check if the user that made the request is the owner of the sentence.

I hear what you say about the notes, so I will be removing the [permission check on the note creation](https://github.com/MetaCell/sckan-composer/pull/319/files#diff-3b354d54e554e0e346c71162641f494dd6a04f3071a51e0851bc9399d1d466a9R325)

File: backend/composer/api/permissions.py [hunk 2]
```diff
@@ -1,9 +1,108 @@
...
+    """
+    sentence_id = request.data.get('sentence_id')
+    connectivity_statement_id = request.data.get('connectivity_statement_id')
+
+    # Check ownership for sentence_id
+    if sentence_id:
+        try:
+            sentence = Sentence.objects.get(id=sentence_id)
+        except Sentence.DoesNotExist:
+            raise PermissionDenied("Invalid sentence_id.")
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply backend/composer/api/permissions.py:95]: Sounds fair. Will make the changes

**Comment by afonsobspinto** [comment]
I'm a bit late to the party but I just now had the chance to review @Salam-Dalloul 's PR. I'm linking the comments I left there, here:
1. https://github.com/MetaCell/sckan-composer/pull/342#pullrequestreview-2407872918
2. https://github.com/MetaCell/sckan-composer/pull/342/files#r1824422388
3.

**Comment by ddelpiano** [comment]
> I'm a bit late to the party but I just now had the chance to review @Salam-Dalloul 's PR. I'm linking the comments I left there, here:
> 
>     1. [check ownership before performing the update #342 (review)](https://github.com/MetaCell/sckan-composer/pull/342#pullrequestreview-2407872918)
> 
>     2. https://github.com/MetaCell/sckan-composer/pull/342/files#r1824422388
> 
> 
> cc @ddelpiano

I believe one of the remark has been resolved already by the last PR, the second one can be a separate PR since I would like to start the testing.

**Review by ddelpiano — APPROVED** [review]
